src/QueueReceiver.Core/Services/PersonService.cs

```diff
@@ -14,26 +14,44 @@ public class PersonService : IPersonService
         private readonly IGraphService _graphService;
         private readonly IProjectRepository _projectRepository;
         private readonly ILogger<PersonService> _logger;
+        private readonly IPersonProjectRepository _personProjectRepository;
 
         public PersonService(
-            IPersonRepository personRepository, 
-            IGraphService graphService, 
+            IPersonRepository personRepository,
+            IGraphService graphService,
             IProjectRepository projectRepository,
+            IPersonProjectRepository personProjectRepository,
             ILogger<PersonService> logger)
         {
             _personRepository = personRepository;
             _graphService = graphService;
             _projectRepository = projectRepository;
+            _personProjectRepository = personProjectRepository;
             _logger = logger;
         }
 
-        public async Task VoidPersonAsync(long personId)
+
+        public async Task UpdateVoidedStatus(string personOid)
+        {
+            var personId = await _personRepository.FindPersonIdByUserOidAsync(personOid);
```

Comment:
At this point, if the Reconcile logic has been hit, there will not be a user stored in the database. The subsequent logic might fail as a result (personId would be NULL or 0). Perhaps a null-check is in order?